### ISSUE TITLE
[Question] Reduce SNMP load for some devices

### ISSUE THREAD

**Issue by jjess** [toni-moreno/snmpcollector#434]
Hi:

I'm dealing with an issue related to some kind heavy SNMP load for device's processors. The general picture is that I'm using SNMPCollector to retrieve some metrics from specific interfaces (very very few) for a set of devices. The devices are Huawei PTNs. 

For some reason, probably related to how I configure the metrics in SNMPCollector, some devices refuse to respond the SNMP requests, so I'm loosing a lot of samples. We have another SNMP monitoring system (vendor solution); when stopping the gathering process for one of these devices then the vendor's solution start to work well, drawing almost all of the samples.

So my question is how to reduce the load for the device when SNMPCollector is polling. 

Some details about how these devices are configured in SNMPCollector:

OID Conditions:
![imagen](https://user-images.githubusercontent.com/902573/82892391-a3f1ff80-9f4f-11ea-8333-76e649fdd1ed.png)
One device can have one or more OID Conditions. 

SNMP Metrics:
![imagen](https://user-images.githubusercontent.com/902573/82892001-f7178280-9f4e-11ea-960b-63666ffcedea.png)

Influx Measurements
![imagen](https://user-images.githubusercontent.com/902573/82892083-1f9f7c80-9f4f-11ea-8d11-32d12672968e.png)
Every device can have several of these influx measurements. Every one of these measurements has a different id and tag. The idea is to separate metrics regarding the kind of traffic each interface carries out.

Measurement Groups
![imagen](https://user-images.githubusercontent.com/902573/82892233-68efcc00-9f4f-11ea-95f4-0d04e3e32f00.png)
Basically, one measurement group per Influx Measurement

Measurement Filters
![imagen](https://user-images.githubusercontent.com/902573/82892324-90df2f80-9f4f-11ea-84a6-328fc5d3c97c.png)

SNMP Devices
![imagen](https://user-images.githubusercontent.com/902573/82892465-ca179f80-9f4f-11ea-8277-f96f1610ea88.png)
![imagen](https://user-images.githubusercontent.com/902573/82892488-d3a10780-9f4f-11ea-9f14-8a177c40810e.png)

What should I do in a different way in order to reduce the SNMP load for the device? Maybe reduce the measurementgroups? but then how to differentiate (tag) between different traffic classes? 

Thanks in advance,

BR, 

Jes

**Comment by jjess** [comment]
In this document:

https://github.com/toni-moreno/snmpcollector/wiki/Component:-Measurements

there is this note:

Notes: We recommend to set all indexed metrics on a same measurement, this will reduce the polling time to create the index or filters.

So, maybe I use only one measurement in every PTN, the snmp load would be lower. But in that case I cannot differentiate between different kinds of traffic. I want to discrimine one kind of traffic (let's say "mobile") from another (let's say "fixed"). Using different "measurements" what I get at the end is separated tables in InfluxDB, and by this choose in Grafana what I want selecting from one table or from another. 

I think I cannot use tags, because I cannot differentiate metrics collected from interfaces carrying out different kind of traffic.

Any idea?

Thanks in advance,

Jes

**Comment by sbengo** [comment]
Hi @jjess ,

Measurements and Filters are using different polling time, based on Freq Filter, so it is not issuing directly your runtime load

Saying that, I think your big load comes from your metrics: when you configure a metric with an indexed measurement the poll is being made with a SNMPWalk and, according to your configuration, you are doing the same query N times, based on num of measurements

I think the only approach for your usecase is the approach proposed on #430  , have you tried it?

On the other hand, you can play with some `Device Polling` configurations:

- Try to decrease/increase `Max Repetitions`, it may increase the polling time but the load done by SNMPWalks will be reduced. IE: 20
- *THIS CAN HEAVILY INCREASE YOUR POLLING TIME* - By default, each measurement is being polled in concurrent mode (pallelism), you can set `ConcurrentGather` to false to make the polling in sequential way, it can reduce your load on device but it will increase your polling time

Thanks,
Regards!

**Comment by jjess** [comment]
Hi:

I couldn't try https://github.com/toni-moreno/snmpcollector/issues/430 because we cannot distinguish the traffic type based on the ifAlias. So I'm stuck. To use ifAlias then I should update the text in every single interface to be monitorized... 

Maybe if the InfluxDB table where the values are stored was Measurement Group ID instead of Measurement ID the issue could be fixed. But obviously this is not possible. 

I have to think on this.

Thanks for your help sbengo.

BR

**Comment by sbengo** [comment]
Hi @jjess ,

I recommend you to play around with polling device settings to see if it reduces the load on your use case.

Closing this issue, we can keep discussing on #430 to see if you can improve your setup for your usecase!

Thanks,
Regards
